client/src/views/Home.vue

```diff
@@ -80,15 +81,13 @@ export default {
     },
   },
   created() {
-    this.location_ = getLocationFromRoute(this.$route);
-    this.setLocation(this.location_);
+    this.setLocation(getLocationFromRoute(this.$route));
     this.notificationBus.$on('message:job_status', this.handleNotification);
   },
   beforeDestroy() {
     this.notificationBus.$off('message:job_status', this.handleNotification);
   },
   beforeRouteUpdate(to, from, next) {
-    this.location_ = getLocationFromRoute(to);
     next();
   },
```

Comment:
Looks like you can remove this whole function.  `next()` is the default behavior.

FWIW, if you needed to do something like this, `vuex-router-sync` is what you want.  This is basically what that plugin is designed for.